File: BACKEND/facturador_backend/src/main/java/com/bap/api/services/api/ApiPuntoVentaService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.bap.api.services.api;

import com.bap.api.configurate.AbstractEntity;
import com.bap.api.dto.Respuesta;
import com.bap.api.dto.SolicitudCliente;
import com.bap.api.model.api.ApiPuntoVenta;
import java.util.List;

/**
 *
 * @author ruth
 */
public interface ApiPuntoVentaService extends AbstractEntity<ApiPuntoVenta> {

    List<ApiPuntoVenta> listarPorIdEmpresa(Long idEmpresa);

    List<ApiPuntoVenta> listarPorIdSucursal(Long IdSucursal);

    Respuesta registroPuntoVenta(SolicitudCliente t);
    
    Respuesta consultaPuntoVenta(SolicitudCliente t);
    
    Respuesta cierrePuntoVenta(SolicitudCliente t);

    ApiPuntoVenta leerPorIdSucursalAndCodigoPuntoVenta(Long idSucursal, Long codigoPuntoVenta);
}

File: FRONTEND/facturador-frontend/src/services/ApiPuntoVentaService.js
import Api from '@/services/Api'

export default {
    ///ApiPuntoVenta
    getApiPuntoVenta(){
        return Api().get("/api/apiPuntoVenta");
    },
    getApiPuntoVentaPorIdEmpresa(idEmpresa){
        return Api().get(`/api/apiPuntoVenta/listaPorIdEmpresa/${idEmpresa}`);
    },
    getApiPuntoVentaPorIdSucursal(idSucursal){
        return Api().get(`/api/apiPuntoVenta/listarPorIdSucursal/${idSucursal}`);
    },
    putApiPuntoVenta(dato){
        return Api().put("/api/apiPuntoVenta", dato);
    },
    registrarPuntoVenta(dato){
        return Api().post("/api/apiPuntoVenta/registrarPuntoVenta", dato);
    },
}
